hngprojects/hng_boilerplate_csharp_web
# feat: added metrics endpoint for prometheus monitoring

# Description
I edited the `Program.cs` file to add the Prometheus package, and make it export server metrics on the `/metrics` endpoint. This change allows our application to expose metrics that can be scraped by Prometheus, enabling better monitoring and observability of our system.

# Changes proposed
## What were you told to do?
Integrate application monitoring with Prometheus and Grafana into our existing C# application

## What did you do?
- Installed the `prometheus-net.AspNetCore` NuGet package
- Modified `Program.cs` to include Prometheus configuration:
  - Added a custom Counter metric for tracking request counts
  - Configured Prometheus middleware to expose metrics at the `/metrics` endpoint
  - Added custom middleware to increment the request count for each request

# Check List (Check all the applicable boxes)

- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
N/A - This change doesn't involve UI modifications. The metrics can be viewed by accessing the `/metrics` endpoint of the application.

To verify the changes:
1. Run the application
2. Access `http://<your-domain-or-ip>:<your_port>/metrics`
3. You should see Prometheus metrics output, including the custom `request_count` metric
